### ISSUE TITLE
LLT doesn't report clock speed if it is above base clock (3.8 ghz on Ryzen 7 7840HS)

### ISSUE THREAD

**Issue by newbietechnologyenthusiast** [BartoszCichecki/LenovoLegionToolkit#1117]
### Rules

- [x] I made myself familiar with the README.
- [x] I checked closed issues for similar ones and did not find a similar problem mentioned before.
- [x] I understand that, if insufficient information was provided, my issue will be closed without an answer.

### Version

2.18.0

### OS

Windows 11 Home 23H2 Build 22631.3007

### Device

Legion Slim 5 16APH8

### BIOS version

M3CN39WW

### What's wrong?

Whenever clock speeds go past the rated base clock of my CPU @3.8ghz, LLT does not report this and instead shows a dash in where the clock speed is supposed to show. 
<img width="474" alt="image" src="https://github.com/BartoszCichecki/LenovoLegionToolkit/assets/108675841/ced1cb55-f76a-4fd0-bc51-277937cecd76">

### How to reproduce the bug?

Just let the clock speed go past base clock.

### What is the behavior that you expected?

The clock speed will be reported 

### Logs

[log_2024_01_10_07_57_55.txt](https://github.com/BartoszCichecki/LenovoLegionToolkit/files/13884707/log_2024_01_10_07_57_55.txt)

**Comment by jimmychuichui** [comment]
I think the program is unable to read the max CPU clock speed, it only know its base clock (3.8GHZ) instead of max clock 5.1GHZ
<img width="772" alt="image" src="https://github.com/BartoszCichecki/LenovoLegionToolkit/assets/60454148/3611a835-cfed-4094-bc60-6e3fc6104e59">

**Comment by BartoszCichecki** [comment]
Max Core Clock reported by Lenovo WMI in this model is 3800 MHz. If it is higher, then Lenovo needs to fix it.

So what happens is that using base clock and perfomance counters CPU core speed might be calculated higher than 3.8GHz (which would be correct), but since max core clock is badly reported as only 3800, progress bar disappears. Might be worth looking into line 124.

**Comment by jimmychuichui** [comment]
> Max Core Clock reported by Lenovo WMI in this model is 3800 MHz. If it is higher, then Lenovo needs to fix it.
> 
> So what happens is that using base clock and perfomance counters CPU core speed might be calculated higher than 3.8GHz (which would be correct), but since max core clock is badly reported as only 3800, progress bar disappears. Might be worth looking into line 124.

Is it only happens on this model or it breaks for all Legion models?

**Comment by jimmychuichui** [comment]
> So far, looks like this model only. No issues here with base clock of 2.3 and max clock of 5.1.

If the issue is only limit to this model, one way to fix that is we manually insert 5100mhz as the max if we know the user is using this model

**Comment by BartoszCichecki** [comment]
No, settings this manually is not an option, because such solution isn't maintainable. My hunch is that this issue might be common to all AMD models, but this needs to be verified.

**Comment by Tallulah88** [comment]
I have this same issue on my Legion Slim 7 16APH8 (it also has the 7840HS CPU).

**Comment by Ace-Radom** [comment]
Will it break sth else when we just let `GetCpuCoreClock` function to return a value larger than `cpuMaxCoreClock`?

https://github.com/BartoszCichecki/LenovoLegionToolkit/blob/e3131cef70b41810c691abb5a0a0010938dd67d5/LenovoLegionToolkit.Lib/Controllers/Sensors/AbstractSensorsController.cs#L119-L126

I don't know if it is okay, when we just re-write this if cause with:

```csharp
if (clock < 1)
    return -1;
```

In this case a CPU core clock higher than the *normal* max CPU core clock can also be returned, and it should actually not break the frontend output.

-----------------

I need to say that I cannot reproduce this bug on my laptop, but I have a similar problem. When I enable Custom Power Mode with Max-Fan-Speed enabled, the fan speed goes actually up to 6300 rpm, which the max fan speed read from `WMI.LenovoFanMethod.GetDefaultFanMaxSpeedAsync` is 4500. The `_cpuFanSpeedLabel` shows 6300RPM as normal, and the `_cpuFanSpeedBar` doesn't break.
